client/src/main/scala/io/delta/sharing/client/DeltaSharingFileSystem.scala

```diff
@@ -69,7 +70,31 @@ private[sharing] class DeltaSharingFileSystem extends FileSystem with Logging {
       val proxy = new HttpHost(proxyConfig.host, proxyConfig.port)
       clientBuilder.setProxy(proxy)
 
-      if (proxyConfig.noProxyHosts.nonEmpty) {
+      val neverUseHttps = getConf.getBoolean("spark.delta.sharing.never.use.https", false)
```

Comment:
So we are only doing https degrading if proxy is active? Looking at the other names of the DS confs, maybe `spark.delta.sharing.network.never.use.https`?

Reply:
yes, only downgrading if proxy is provided. If this is an unnecessary restriction, i can remove.